internal/consensus/reactor.go

```diff
@@ -665,6 +668,22 @@ OUTER_LOOP:
 			continue OUTER_LOOP
 		}
 
+		// --------------------
+		// Send blob part?
+		// (Note these can match on hash so round doesn't matter)
+		// --------------------
+
+		if part, continueLoop := pickBlobPartToSend(rs, prs, rng); part != nil {
+			// part is not nil: we either succeed in sending it,
+			// or we were instructed not to sleep (busy-waiting)
+			if ps.SendPartSetHasPart(part, prs) || continueLoop {
```

Comment:
This cannot be reused as `SendPartSetHasPart` marks the block parts as sent.

Reply:
Uh, good catch! I'll look into it.